ambry-tools/src/main/java/com.github.ambry/store/DumpIndexTool.java

```diff
@@ -220,21 +246,29 @@ public long getDeleteCount() {
       return updateAfterDeletes;
     }
 
+    /**
+     * @return the throwable set (if any)
+     */
+    public Throwable getThrowable() {
+      return throwable;
+    }
+
     /**
      * @return {@code true} if there are no duplicate puts or deletes or updates and no puts after updates or updates
      * after deletes. {@code false} otherwise
      */
     public boolean isIndexSane() {
-      return duplicatePuts.size() == 0 && putAfterUpdates.size() == 0 && duplicateDeletes.size() == 0
-          && duplicateUpdates.size() == 0 && updateAfterDeletes.size() == 0;
+      return throwable == null && duplicatePuts.size() == 0 && putAfterUpdates.size() == 0
+          && duplicateDeletes.size() == 0 && duplicateUpdates.size() == 0 && updateAfterDeletes.size() == 0;
     }
 
     @Override
     public String toString() {
-      return "Processed: " + processedCount + ", Active: " + activeCount + ", Put: " + putCount + ", TTL update count: "
-          + ttlUpdateCount + ", Delete: " + deleteCount + ", Duplicate Puts: " + duplicatePuts + ", Put After Update: "
-          + putAfterUpdates + ", Duplicate Delete: " + duplicateDeletes + ", Duplicate Update: " + duplicateUpdates
-          + ", Update After Delete: " + updateAfterDeletes;
+      return throwable == null ? "Processed: " + processedCount + ", Active: " + activeCount + ", Put: " + putCount
+          + ", TTL update count: " + ttlUpdateCount + ", Delete: " + deleteCount + ", Crafted ID count: "
+          + craftedIdCount + ", Duplicate Puts: " + duplicatePuts + ", Put After Update: " + putAfterUpdates
+          + ", Duplicate Delete: " + duplicateDeletes + ", Duplicate Update: " + duplicateUpdates
+          + ", Update After Delete: " + updateAfterDeletes : "Exception Msg: " + throwable.getMessage();
```

Comment:
I am thinking if we could make this more readable by using StringBuilder to append. Something like this:
```
sb.append("Processed: ").append(processedCount)
  .append(", Active: ").append(activeCount)
...
```

Reply:
Formatted this to be more readable. Please take a look